src/libmongoc/tests/unified/entity-map.c

```diff
@@ -561,6 +563,11 @@ session_opts_new (bson_t *bson, bson_error_t *error)
    if (causal_consistency) {
       mongoc_session_opts_set_causal_consistency (opts, *causal_consistency);
    }
+   if (snapshot) {
+      // Set causal consistency to false to allow snapshot
+      mongoc_session_opts_set_causal_consistency (opts, false);
```

Comment:
I don't think this is necessary. Since the combination of causal consistency and snapshot reads is invalid, we shouldn't quietly fix this scenario in the tests but forbid it elsewhere. A test specifying both causal consistency and snapshot reads should fail so the offending test is highlighted instead of silently being fixed.

Reply:
There is an [ongoing discussion](https://github.com/mongodb/specifications/pull/1022#discussion_r657999964) in the new specification about whether to turn off `causalConsistency` when `isSnapshot` is set. `causalConsistency` is true by default, so without this line we'd technically error when attempting to `set_snapshot` to true.

Reply:
The alternative is that `mongoc_session_opts_set_snapshot` could disable causal consistency. But I think the current behavior aligns with the spec:

> If no value is provided for ``causalConsistency`` and snapshot reads are not requested a value of true is implied. See the ``causalConsistency`` section.

`mongoc_session_opts_new` always creates options with causal consistency enabled since it does not accept arguments.